web-marketplace/src/components/organisms/ChooseCreditsForm/ChooseCreditsForm.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  ChangeEvent,
+  MouseEvent,
+  Suspense,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { SubmitHandler, useWatch } from 'react-hook-form';
+import { CreditsAmount } from 'web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount';
+import {
+  CREDITS_AMOUNT,
+  CURRENCY_AMOUNT,
+} from 'web-marketplace/src/components/molecules/CreditsAmount/CreditsAmount.constants';
+import Form from 'web-marketplace/src/components/molecules/Form/Form';
+import { useZodForm } from 'web-marketplace/src/components/molecules/Form/hook/useZodForm';
+
+import Card from 'web-components/src/components/cards/Card';
+import {
+  CURRENCIES,
+  Currency,
+} from 'web-components/src/components/DenomIconWithCurrency/DenomIconWithCurrency.constants';
+import { Loading } from 'web-components/src/components/loading';
+
+import { AdvanceSettings } from './ChooseCreditsForm.AdvanceSettings';
+import { PAYMENT_OPTIONS } from './ChooseCreditsForm.constants';
+import { CryptoOptions } from './ChooseCreditsForm.CryptoOptions';
+import { PaymentOptions } from './ChooseCreditsForm.PaymentOptions';
+import {
+  ChooseCreditsFormSchemaType,
+  createChooseCreditsFormSchema,
+} from './ChooseCreditsForm.schema';
+import {
+  CreditDetails,
+  CreditsVintages,
+  PaymentOptionsType,
+} from './ChooseCreditsForm.types';
+import { getSpendingCap } from './ChooseCreditsForm.utils';
+
+export function ChooseCreditsForm({
+  creditVintages,
+  creditDetails,
+}: {
+  creditVintages: CreditsVintages[];
+  creditDetails: CreditDetails[];
```

Comment:
After second thoughts, I believe we're not quite there yet in terms of props because the amount of credits available per vintage can change based on the selected currency.
When a sell order is created, you pick up a vintage (=batch) and choose a price denom. Meaning that for a same vintage, you could have different sell orders with different ask prices and denoms.

Reply:
Unless as soon as the currency changes, we propagate it to the parent component, which will then update the available `creditVintages`...
The other option would simply append to each `creditDetails` a list of available `creditVintages` and the sum of those vintages credits would be equal to the `creditDetails.availableCredits`
For crypto purchase, we also need to know whether sold credits are tradable or not, because if the user picks up "Buy tradable ecocredits" option then we don't want to show credits for sell that are not tradable

Reply:
Agree with doing it when we work on the integration.

Reply:
alright, let's do that but let's at least address the following issue: https://github.com/regen-network/regen-web/pull/2408#issuecomment-2331367225

Reply:
and add a comment in the code as a TODO about what I said above